docs/auth/public_sharing_of_statistics.md

```diff
@@ -100,3 +100,108 @@ From [Slack Canvas](https://boxwise.slack.com/docs/T99PBKNTU/F06QQS70XQT):
     4. need to generate many endpoints for large data sets
     5. OR: more open /organisation/<id> endpoint but needs server-side permissioning
 6. consider rate-limiting or DDoS protection
+
+
+# Draft for prototypical implementation
+
+The following serves as a concept for a first implementation of the "link-sharing" functionality.
+
+## Idea
+
+1. Authenticated user of base X creates link for statviz page
+1. Authenticated user shares link with external person
+1. External person opens link
+1. External person sees statviz page for base X without any navigation options
+1. After one week, the link expires. When the link is opened, a short message is displayed.
+
+## Architecture
+
+### Fundamental assumptions
+
+- the shared data will be live (as opposed to "freezing" the data to the time of link creation)
+- link format is `.../<code> (can't use `.../bases/X/statviz`: insecure because it's easy to navigate to other bases and view their data; also not unique (can't expire))
+- the expiration time of the link is one week. Later we can make is customizable, and/or add an action to invalidate a created link
+- we expose the public app in the `api` GAE service. Hence it won't interfer with the main `app` service
+
+### Effects on full-stack
+
+- there has to be a new action-based permission to allow link-creation for certain usergroups
+- there has to be a public FE and a public GraphQL endpoint to fetch data
+- the public FE is hosted under `api.boxtribute.org/shared` and deployed with the `deploy-api` CircleCI job (depends a new `build-statviz` job)
```

Comment:
I suggested this for simplicity (less devops effort because we don't need to establish a new subdomain nor any new GAE services). The endpoints we currently have for the `api` subdomains are the GraphQL one at `/` and the documentation at `/docs`.
For a separation of concerns it probably makes sense to have a dedicated subdomain for link-sharing.

Reply:
@pylipp got it. Is there any other way we can achieve that? If the back end link sharing is going to be part of the standard API on the api subdomain, then any reason the link sharing can't just operate as part of the same FE code base? Or am I misunderstanding here?

Reply:
Sorry, this comment slipped through somehow. @jamescrowley 

> any reason the link sharing can't just operate as part of the same FE code base? 
Do you want the link-sharing back-end to be part of the FE code base? Or do you want the public FE for shared links to be in the same code base as the standard FE?

One more thing re the new public BE: When we introduce a new `/shared` route on the api subdomain, we can (and will) use a new schema there, different from the standard API schema (hosted on `/`).